In [65]:
import pandas as pd

df_train = pd.read_csv("train.csv")
df_test = pd.read_csv("test.csv")

In [66]:
df_train["ID"].head(10) # The target column is to be predicted
                 # Do I need to keep the ID? No real info brought?
    
print 'The shape of my Dataframe is {0} x {1}' .format(df_train.shape[0],df_train.shape[1])

df_train.describe() # all the data counts for 114,321


The shape of my Dataframe is 114321 x 133


ID         target            v1            v2            v4  \
count  114321.000000  114321.000000  6.448900e+04  6.452500e+04  6.452500e+04   
mean   114228.928228       0.761199  1.630686e+00  7.464411e+00  4.145098e+00   
std     65934.487362       0.426353  1.082813e+00  2.961676e+00  1.148263e+00   
min         3.000000       0.000000 -9.996497e-07 -9.817614e-07 -6.475929e-07   
25%     57280.000000       1.000000  9.135798e-01  5.316428e+00  3.487398e+00   
50%    114189.000000       1.000000  1.469550e+00  7.023803e+00  4.205991e+00   
75%    171206.000000       1.000000  2.136128e+00  9.465497e+00  4.833250e+00   
max    228713.000000       1.000000  2.000000e+01  2.000000e+01  2.000000e+01   

                 v5            v6            v7            v8            v9  \
count  6.569700e+04  6.448900e+04  6.448900e+04  6.570200e+04  6.447000e+04   
mean   8.742359e+00  2.436402e+00  2.483921e+00  1.496569e+00  9.031859e+00   
std    2.036018e+00  5.999653e-01  5.894485e-01  2.783003e+00  1.930262e+00   
min   -5.287068e-07 -9.055091e-07 -9.468765e-07 -7.783778e-07 -9.828757e-07   
25%    7.605918e+00  2.065064e+00  2.101477e+00  8.658986e-02  7.853659e+00   
50%    8.670867e+00  2.412790e+00  2.452166e+00  3.860317e-01  9.059582e+00   
75%    9.771353e+00  2.775285e+00  2.834285e+00  1.625246e+00  1.023256e+01   
max    2.000000e+01  2.000000e+01  2.000000e+01  2.000000e+01  2.000000e+01   

           ...               v121          v122          v123          v124  \
count      ...       6.448100e+04  6.447000e+04  63643.000000  6.570200e+04   
mean       ...       2.737596e+00  6.822439e+00      3.549938  9.198120e-01   
std        ...       1.356294e+00  1.795978e+00      2.604704  2.099407e+00   
min        ...      -9.820642e-07 -9.978497e-07      0.019139 -9.994953e-07   
25%        ...       1.786965e+00  5.647712e+00      1.963315  2.053777e-02   
50%        ...       2.436195e+00  6.749117e+00      2.739239  1.398639e-01   
75%        ...       3.379175e+00  7.911392e+00      4.075361  8.718333e-01   
max        ...       2.000000e+01  2.000000e+01     19.686069  2.000000e+01   

               v126          v127          v128           v129          v130  \
count  6.448900e+04  6.448900e+04  6.569700e+04  114321.000000  6.447800e+04   
mean   1.672658e+00  3.239542e+00  2.030373e+00       0.310144  1.925763e+00   
std    5.031683e-01  1.625988e+00  1.074232e+00       0.693262  1.264497e+00   
min   -9.564174e-07 -9.223798e-07  8.197812e-07       0.000000 -9.901257e-07   
25%    1.417600e+00  2.101900e+00  1.393830e+00       0.000000  1.106172e+00   
50%    1.614802e+00  2.963620e+00  1.798436e+00       0.000000  1.560138e+00   
75%    1.843886e+00  4.108146e+00  2.390158e+00       0.000000  2.332425e+00   
max    1.563161e+01  2.000000e+01  2.000000e+01      11.000000  2.000000e+01   

               v131  
count  6.442600e+04  
mean   1.739389e+00  
std    1.134702e+00  
min   -9.999134e-07  
25%    1.012658e+00  
50%    1.589403e+00  
75%    2.261905e+00  
max    2.000000e+01  

[8 rows x 114 columns]

In [70]:
## Preprocessing ##

# The columns v107 and v90 are the same

target = df_train["target"]

try:
    df_train.drop(labels= ["target","ID","v107"],axis=1,inplace=True)
except:
    pass

try:
    df_test.drop(labels = ["ID","v107"],axis=1,inplace=True)
except:
    pass


0         1
1         1
2         1
3         1
4         1
5         0
6         0
7         1
8         0
9         1
10        1
11        1
12        0
13        1
14        1
15        1
16        1
17        1
18        1
19        1
20        1
21        1
22        1
23        1
24        1
25        1
26        1
27        1
28        1
29        0
         ..
114291    1
114292    1
114293    1
114294    1
114295    1
114296    1
114297    1
114298    1
114299    1
114300    0
114301    1
114302    1
114303    1
114304    1
114305    1
114306    1
114307    1
114308    1
114309    0
114310    1
114311    1
114312    1
114313    1
114314    0
114315    1
114316    1
114317    1
114318    1
114319    1
114320    1
Name: target, dtype: int64

In [71]:
# Using the data from the test set
# We will extract the rows which have no NaN values

df_test.describe() # all the data counts for 114,321

print 'The shape of my Dataframe is {0} x {1}' .format(df_test.shape[0],df_test.shape[1])

x_test = df_test.isnull().values

# print(x[0:5])

x_test = x_test.sum(axis=1)

extract_full_rows = (x_test == 0)

full_data = df_test.loc[extract_full_rows,:]
print(" Size of the full data DataFrame {0} x {1}".format(full_data.shape[0], full_data.shape[1]))


# Number of rows with n non null values
# Could it be possible to use it as a feature 
# for Machine Learning?

x_test = pd.Series(x_test)
count_df_test = pd.DataFrame(x_test.value_counts())

count_df_test = count_df_test.reset_index()
count_df_test.columns = ["Number of null values per row","Count of Rows"]
count_df_test = count_df_test.sort_values(["Number of null values per row"], ascending=True)
count_df_test


The shape of my Dataframe is 114393 x 130
 Size of the full data DataFrame 17651 x 130


Number of null values per row  Count of Rows
2                               0          17651
0                               1          28487
4                               2          14308
6                               3           1747
7                               4           1335
20                              5             34
17                              6             43
25                              7             15
26                              8              9
33                              9              2
36                             10              1
37                             11              1
30                             12              5
29                             13              5
45                             16              1
38                             17              1
39                             18              1
10                             25            583
13                             26            249
21                             27             29
32                             28              3
43                             30              1
35                             31              2
44                             36              1
28                             64              8
22                             65             22
27                             66              8
40                             68              1
12                             81            453
8                              82           1077
11                             83            458
18                             84             41
19                             85             40
41                             88              1
42                             89              1
5                             100           4338
1                             101          27038
3                             102          15269
9                             103            720
14                            104            238
15                            105             78
24                            106             16
16                            107             47
23                            108             20
34                            109              2
31                            110              3

In [222]:
df_train["v110"].head(10)

df_types = df_train.dtypes

# My non numerical values
non_num_columns = df_types[df_types == "object"].index

# My numerical values
num_columns = df_types[df_types != "object"].index

df_non_num = df_train.loc[:,non_num_columns]

# pd.concat([df_non_num,df["target"]],axis=1)
# df["v22"].value_counts()
df_non_num
y = df_non_num["v113"].unique()
y = pd.DataFrame(y)

y.columns = ["label"]
y.loc[y["label"] =="I"]


ref_dict = filter(lambda x: len(x)==1,y.values.flatten()[1:])
len(filter(lambda x: len(x)==1,pd.DataFrame(df_test["v113"].unique()).values.flatten()[1:]))

#def label_encoding(label):
#    
#    ref_dict
#    parsing_list = list(label)
#    if len(parsing_list)>1:
        
#y    
    
    


26

In [131]:
list("Hello")

['H', 'e', 'l', 'l', 'o']

In [11]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline


def turn_categorical(x):
    
    dict_v3 = {"A":1,"B":2,"C":3}
    try:
        return dict_v3[x]
    except:
        pass

# Let's compute the number of intances which are missing per field

# df["v2"].apply(np.isnan).sum()

df_train.loc[df_train["v1"].apply(np.isnan)].head(10)

# Let's see how v3 categorical value correlate with the target values
# What kind of calues do I have in v3:

df_train["v3"].value_counts() # A : 227
                        # B : 53
                        # C : 110,584
        


serie_target = pd.Series(df_train["v3"].map(lambda x: turn_categorical(x)))
serie_target


#plt.scatter(serie_target,df[""])

x = df_train.isnull().values

# print(x[0:5])

x = x.sum(axis=1)

extract_full_rows = (x == 0)

full_data = df_train.loc[extract_full_rows,:]
print(" Size of the full data DataFrame {0} x {1}".format(full_data.shape[0], full_data.shape[1]))


# Number of rows with n non null values
# Could it be possible to use it as a feature 
# for Machine Learning?

x = pd.Series(x)
count_df = pd.DataFrame(x.value_counts())

count_df = count_df.reset_index()
count_df.columns = ["Number of null values per row","Count of Rows"]
count_df = count_df.sort_values(["Number of null values per row"], ascending=True)
# count_df


 Size of the full data DataFrame 17756 x 133


Only 17756 rows have complete data. I will run my the theano NN on it to fill the other ones?

In [44]:
import numpy as np
# print(df_train["v90"].value_counts())

# print(df_train["v107"].value_counts())

# The features v91 and v107 are the same, oly diffrent levels. we need to drop one of the two.
# Dropped the v107 columns and the Target one


df_train

ID            v1         v2   v3        v4         v5        v6  \
0            3  1.335739e+00   8.727474    C  3.921026   7.915266  2.599278   
1            4           NaN        NaN    C       NaN   9.191265       NaN   
2            5  9.438769e-01   5.310079    C  4.410969   5.326159  3.979592   
3            6  7.974146e-01   8.304757    C  4.225930  11.627438  2.097700   
4            8           NaN        NaN    C       NaN        NaN       NaN   
5            9           NaN        NaN    C       NaN   8.856791       NaN   
6           12  8.998057e-01   7.312995    C  3.494148   9.946200  1.926070   
7           21           NaN        NaN    C       NaN        NaN       NaN   
8           22  2.078651e+00   8.462619  NaN  3.739030   5.265636  1.573033   
9           23  1.144802e+00   5.880606    C  3.244469   9.538384  2.500001   
10          24           NaN        NaN    C       NaN        NaN       NaN   
11          27           NaN        NaN    C       NaN        NaN       NaN   
12          28           NaN        NaN    C       NaN        NaN       NaN   
13          30  1.400267e+00   5.367204    C  4.122155   8.137188  2.983080   
14          31  2.260036e+00  14.693263    C  5.150750   8.554136  1.954626   
15          32           NaN        NaN    C       NaN        NaN       NaN   
16          33  6.228961e-01   7.024732    C  4.193688   6.288177  2.132436   
17          34           NaN        NaN    C       NaN        NaN       NaN   
18          35           NaN        NaN    C       NaN        NaN       NaN   
19          36           NaN        NaN    C       NaN        NaN       NaN   
20          37  9.438780e-01   5.927194    C  4.404372   9.045057  2.551021   
21          39  1.289841e+00   4.788645    C  4.283417  10.719571  3.107570   
22          40  7.288239e-01   4.073244    C  4.130054   9.032563  4.149705   
23          42  3.944563e+00   5.718516    C  2.205080   5.340648  2.010356   
24          43  4.045725e+00   3.992607    C  3.598096   7.946330  1.709742   
25          46           NaN        NaN    C       NaN        NaN       NaN   
26          51           NaN        NaN    C       NaN        NaN       NaN   
27          52  2.920282e-01  12.715811    C  4.999724   5.998656  2.020520   
28          54  5.172412e-01   8.528544    C  4.075372   5.726834  2.534949   
29          55  2.415670e+00  14.960392    C  4.356980   8.520510  2.132753   
...        ...           ...        ...  ...       ...        ...       ...   
114291  228655           NaN        NaN    C       NaN        NaN       NaN   
114292  228657  2.768086e-01   7.055455    C  3.823010   9.619069  2.194513   
114293  228658  1.983914e+00  11.296659    C  6.459140   9.193006  1.849866   
114294  228659           NaN        NaN    C       NaN        NaN       NaN   
114295  228660  8.743610e-01   8.529843    C  4.391163   9.575062  2.646711   
114296  228663           NaN        NaN    C       NaN        NaN       NaN   
114297  228666  1.603274e+00  11.493653    C  4.328798   8.055434  1.848820   
114298  228668  9.477454e-01   8.941986    C  4.578888  14.882313  2.868852   
114299  228670  9.688237e-07   6.163454    C  3.502739   7.202643  1.741765   
114300  228671  1.539335e+00   4.845475    C  4.445304   8.605211  1.906204   
114301  228673           NaN        NaN    C       NaN        NaN       NaN   
114302  228677  2.093352e+00   5.696141    C  5.105297   8.836584  2.734559   
114303  228680           NaN        NaN    C       NaN        NaN       NaN   
114304  228682           NaN        NaN    C       NaN        NaN       NaN   
114305  228683  1.030641e+00  13.003754    C  4.512371   8.928965  2.548746   
114306  228684  2.051611e+00   8.259554    C  4.466166  11.291191  2.269140   
114307  228688           NaN        NaN    C       NaN        NaN       NaN   
114308  228691           NaN        NaN    C       NaN        NaN       NaN   
114309  228695           NaN        NaN    C       NaN        

In [2]:
import sklearn

ImportError: No module named sklearn